File: Java/PlenoInterview2/Problem1_1/InsertionSorter.java
public class InsertionSorter implements Sorter{

    public InsertionSorter(){}

    public int[] Sort(int[] array){
        System.out.println("Sorting with Insertion Sort.");

        int[] sortedArray = array.clone();

        for (int i=2; i<sortedArray.length-2; i++)
        {
            int j = i;
            while(j>0 && sortedArray[j-1] < sortedArray[j]) {
                int tmp = sortedArray[j];
                sortedArray[j] = sortedArray[j - 1];
                sortedArray[j - 1] = tmp;
                j--;
            }
        }
        return sortedArray;
    }

}

File: Problems/FixTheCode/FTC_3_sorting-algorithms/python/source/insertionSorter.py
from customSorter import CustomSorter
from arrays import swapTwoNumbers

class InsertionSorter(CustomSorter):

    def sort(self):
        print("Sorting with INSERTION SORT.")
        arraySize = len(self.array)

        for i in range(1, arraySize):
            j = i
            while j > 0 and j < arraySize and self.array[j - 1] < self.array[j]:
                swapTwoNumbers(self.array, j, j - 1)
                j -= 2 